x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/cache/PersistentCacheTests.java

```diff
@@ -0,0 +1,193 @@
+                    assertThat("Document should be loaded", document, notNullValue());
+                    final String iteration = document.get("update_iteration");
+                    assertThat(iteration, equalTo(String.valueOf(liveDoc.getValue())));
+                    writer.updateCacheFile(liveDoc.getKey(), document);
+                }
+
+                // verify that deleted documents are not loaded
+                for (String deletedDoc : deletedDocs) {
+                    final Document document = documents.get(deletedDoc);
+                    assertThat("Document should not be loaded", document, nullValue());
+                }
+
+                // random updates of existing documents
+                final Map<String, Integer> updatedDocs = new HashMap<>();
+                for (String cacheId : randomSubsetOf(liveDocs.keySet())) {
+                    final Document document = new Document();
+                    document.add(new StringField("cache_id", cacheId, Field.Store.YES));
+                    document.add(new StringField("update_iteration", String.valueOf(iter), Field.Store.YES));
+                    writer.updateCacheFile(cacheId, document);
+
+                    updatedDocs.put(cacheId, iter);
+                }
+
+                // create new random documents
+                final Map<String, Integer> newDocs = new HashMap<>();
+                for (int i = 0; i < between(1, 10); i++) {
+                    final String cacheId = String.valueOf(docId++);
+                    final Document document = new Document();
+                    document.add(new StringField("cache_id", cacheId, Field.Store.YES));
+                    document.add(new StringField("update_iteration", String.valueOf(iter), Field.Store.YES));
+                    writer.updateCacheFile(cacheId, document);
+
+                    newDocs.put(cacheId, iter);
+                }
+
+                // deletes random documents
+                final Map<String, Integer> removedDocs = new HashMap<>();
+                for (String cacheId : randomSubsetOf(Sets.union(liveDocs.keySet(), newDocs.keySet()))) {
+                    writer.deleteCacheFile(cacheId);
+
+                    removedDocs.put(cacheId, iter);
+                }
+
+                boolean commit = false;
+                if (frequently()) {
+                    writer.prepareCommit();
+                    if (frequently()) {
+                        writer.commit();
+                        commit = true;
+                    }
+                }
+
+                if (commit) {
+                    liveDocs.putAll(updatedDocs);
+                    liveDocs.putAll(newDocs);
+                    for (String cacheId : removedDocs.keySet()) {
+                        liveDocs.remove(cacheId);
+                        deletedDocs.add(cacheId);
+                    }
+                }
+            }
+        }
+    }
+
+    private static final byte[] buffer;
```

Comment:
nit: move to local variable?

Reply:
Sure. I pushed 2940d20433ddf0d9fcfbe7111d2f97112438872a